### DISCUSSION TITLE
macOS cursor (pointer) does not change between terminal area and title bar

### DISCUSSION THREAD

**Discussion by dhowden** [ghostty-org/ghostty#3257] [Issue Triage] [answered]
Would expect the mouse pointer to be the arrow over the titlebar, and the i-beam when over the terminal area.

Instead the pointer seems to be determined by the "first click" that brings the window into focus, and doesn't change as you move between the title/terminal area.

**Comment by mitchellh** [comment]
Can you record a video? I'm sorry I don't understand what you're saying. Can you also share your configuration, since there are a variety of titlebar styles and so on. Thank you!

**Reply by r32pat** [reply]
I might be able to assist here.

If you are on OSX open a new term tab using the COMMAND+T shortcut.
There is a situation where the mouse cursor changes to the Text input style cursor from an arrow when moving over text in the terminal window.

Then when you mouse up to the tab section to say close that tab, the mouse cursor remains Text input style rather than change back to an arrow when leaving the terminal frame.

If you still need a video let me know and ill snap one real quick.

**Reply by dhowden** [reply]
Thanks @r32pat!

> Can you record a video? I'm sorry I don't understand what you're saying.

😂 My bad.  I wrote a few paragraphs originally and thought it was a bit too much, so chopped it down.  Hopefully this repro will help:

Open Ghostty and Terminal.app side-by-side.

Switch focus to Terminal.app by clicking on the titlebar.  Note that when you move the mouse over the title bar it is the arrow, and then you move over the emulator area it chagnes to the I-bar (hinting input), the names from here: https://support.apple.com/en-au/guide/mac-help/mh35695/mac.

Now click in the title bar area of Ghostty to focus the app, and move the mouse between the title bar area and the emulator area.  The cursor remains the Arrow (doesn't switch to I-bar when over the emulator area), it also stays the pointer if you select text etc.   Now click on Terminal.app (to move focus away from Ghostty) and then back to Ghostty by clicking in the emulator area this time.  Now the cursor is only the i-bar (even if you move it up to the title bar).

Hopefully this helps.  I attempted quick recording with QuickTIme but it was messing with the pointers itself :(.

**Reply by yawaramin** [reply]
Hi Mitchell, see https://github.com/ghostty-org/ghostty/discussions/3565 for a screen recording

**Reply by newmana** [reply]
I think this is what they are asking for:
https://developer.apple.com/documentation/appkit/nscursor/ibeam

When hovering over selectable text the cursor becomes the iBeam.

**Comment by sfsam** [comment]
In addition, the mouse pointer should change to the pointing hand shape when hovering over a URL while pressing the Command (Mac) / Control (Linux) key.

**Comment by keysmashes** [comment]
For me, on macOS the mouse pointer remains the default shape regardless of where I position it – I'd expect it to become an I-beam when hovering over the text area, like it does on Linux. (If I set `macos-titlebar-style = tabs`, it does turn into the grabby hand when hovering over the upper half of the tabs.)

...or, rather, it seems like the pointer shape isn't always updated when it should be – with `mouse-hide-while-typing = true`, if I hover over the text area, type `vim` and hit enter, then wiggle the mouse to show the pointer again, it _does_ turn into the I-beam. If I move the mouse onto the lower half of the tab bar (or onto the traffic light window controls), it remains an I-beam; if I move it out of the window and back in, or over the upper half of the tab bar and back, it reverts to the default pointer shape.

**Comment by MatthiasPortzel** [comment]
Just reopened Ghostty today, and remembered why I stopped using it (because of this issue). I'm confused why this isn't a bigger deal (maybe it only effects certain OS versions or something?)—because every other application I've ever opened on my mac has a text-cursor when the text under the cursor is selectable.

**Comment by jhaemin** [comment]
https://github.com/user-attachments/assets/ce38f822-f312-48da-bbae-e34f65eeb6f4

Here is a recording. Cursor doesn't change based on the areas.

It's really annoying.

**Comment by mitchellh** [comment, marked as answer]
Accepted #8409
